components/raftstore/src/store/peer.rs

```diff
@@ -1254,6 +1264,19 @@ where
                     // prewrites or commits will be just a waste.
                     self.last_urgent_proposal_idx = self.raft_group.raft.raft_log.last_index();
                     self.raft_group.skip_bcast_commit(false);
+
+                    // When a peer becomes a leader, update `max_ts_synced` to new term and
+                    // mark it unsynced. Then, update max timestamp from PD asynchronously.
+                    self.max_ts_synced.store(self.term() << 1, Ordering::SeqCst);
+                    if let Err(e) = ctx.pd_scheduler.schedule(PdTask::UpdateMaxTimestamp {
+                        term: self.term(),
+                        max_ts_synced: self.max_ts_synced.clone(),
+                    }) {
+                        error!(
+                            "failed to update max ts";
+                            "err" => ?e,
+                        );
+                    }
```

Comment:
How is the peer notified that it's able to serve after updating max_ts? Or should the user of the snapshot always check if the snapshot's state is `synced`?

Reply:
The state is checked by the user because not all operations are unavailable when unsynced. Only prewrite checks it.